File: baekjoon/9948.cpp
#include<bits/stdc++.h>
using namespace std;
#define loop(i,n) for(int i=0;i<n;++i)

string Month[12] = { "January", "February", "March", "April", "May","June","July","August","September","October","November","December" };
int main()
{
    cin.tie(NULL), cout.tie(NULL);
    ios::sync_with_stdio(false);

    string s;
    int d, m;
    while (cin >> d >> s) {
        if (d == 0) break;
        loop(i, 12)
            if (s == Month[i]) m = i + 1;
        if (m == 2 && d == 29) cout << "Unlucky\n";
        else if (m < 8 || (m == 8 && d < 4)) cout << "Yes\n";
        else if (m == 8 && d == 4) cout << "Happy birthday\n";
        else cout << "No\n";
    }
    return 0;
}

File: baekjoon/9000/9948.rs
use std::io::*;

struct Scanner {
    it: std::str::SplitAsciiWhitespace<'static>,
}

impl Scanner {
    fn new() -> Self {
        let mut s = String::new();
        stdin().read_to_string(&mut s).ok();
        Self { it: s.leak().split_ascii_whitespace() }
    }
    fn read<T: std::str::FromStr>(&mut self) -> T {
        self.it.next().unwrap().parse::<T>().ok().unwrap()
    }
    fn raw(&mut self) -> &str {
        self.it.next().unwrap()
    }
}

fn main() {
    let mut sc = Scanner::new();
    let mut bw = BufWriter::new(stdout().lock());
    loop {
        let d = sc.read::<i32>();
        let m = match sc.raw() {
            "January" => 1,
            "February" => 2,
            "March" => 3,
            "April" => 4,
            "May" => 5,
            "June" => 6,
            "July" => 7,
            "August" => 8,
            "September" => 9,
            "October" => 10,
            "November" => 11,
            "December" => 12,
            _ => break,
        };
        let ans = if m == 2 && d == 29 {
            "Unlucky"
        } else if m < 8 || m == 8 && d < 4 {
            "Yes"
        } else if m == 8 && d == 4 {
            "Happy birthday"
        } else {
            "No"
        };
        writeln!(bw, "{ans}").ok();
    }
}
